watertap/examples/flowsheets/case_studies/BSM2/ASM1_ADM1_flowsheet.py

```diff
@@ -0,0 +1,367 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+__author__ = "Alejandro Garciadiego"
+
+import pyomo.environ as pyo
+from pyomo.environ import (
+    units,
+)
+
+from pyomo.network import Arc, SequentialDecomposition
+from idaes.core import FlowsheetBlock
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+from watertap.unit_models.anaerobic_digestor import AD
+from watertap.unit_models.thickener import Thickener
+from watertap.unit_models.dewatering import DewateringUnit
+
+from watertap.unit_models.translators.translator_asm1_adm1 import Translator_ASM1_ADM1
+from watertap.unit_models.translators.translator_adm1_asm1 import Translator_ADM1_ASM1
+from idaes.models.unit_models import Separator, Mixer
+
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    large_residuals_set,
+    activated_inequalities_set,
+)
+import idaes.logger as idaeslog
+from idaes.core.util.initialization import propagate_state
+from idaes.core.solvers import get_solver
+import idaes.core.util.scaling as iscale
+from idaes.core.util.tables import (
+    create_stream_table_dataframe,
+    stream_table_dataframe_to_string,
+)
+
+from watertap.property_models.anaerobic_digestion.adm1_properties import (
+    ADM1ParameterBlock,
+)
+from watertap.property_models.activated_sludge.asm1_properties import (
+    ASM1ParameterBlock,
+)
+from watertap.property_models.anaerobic_digestion.adm1_reactions import (
+    ADM1ReactionParameterBlock,
+)
+from idaes.models.unit_models.separator import SplittingType
+from watertap.property_models.anaerobic_digestion.adm1_properties_vapor import (
+    ADM1_vaporParameterBlock,
+)
+
+from idaes.core.util.tables import (
+    create_stream_table_dataframe,
+    stream_table_dataframe_to_string,
+)
+
+from watertap.examples.flowsheets.case_studies.activated_sludge.ASM1_flowsheet import (
+    build_flowsheet as build_ASM,
+)
+
+from watertap.core.util.initialization import check_solve
+
+
+def build_flowsheet():
+    # Call ASM model
+    m, res = build_ASM()
+
+    m.del_component(m.fs.Sludge)
+    m.del_component(m.fs.stream1)
+    m.del_component(m.fs.stream102)
```

Comment:
Maybe we should revise what happens under `build_ASM` so that there is no need to delete components; this would be a safer bet. Then, for the use case where these components are not omitted, we can explicitly add them for that/those cases.